pkg/kube/inject/webhook.go

```diff
@@ -1075,8 +1075,12 @@ func (wh *Webhook) inject(ar *kube.AdmissionReview, path string) *kube.Admission
 		injectedAnnotations: wh.Config.InjectedAnnotations,
 		proxyEnvs:           parseInjectEnvs(path),
 	}
-	clusterID, _ := extractClusterAndNetwork(params)
-	if wh.namespaces != nil {
+
+	if platform.IsOpenShift() && wh.namespaces != nil {
+		clusterID, _ := extractClusterAndNetwork(params)
+		if clusterID == "" {
+			clusterID = "Kubernetes"
```

Comment:
Shouldn't this be 

`clusterID = $ISTIO_META_CLUSTER_ID` or equivalent, rather than hardcoded to `Kubernetes`? We already have the concept of a configurable "default" clusterID in many charts/components.

Reply:
The issue is that this env var is empty - it is checked inside `extractClusterAndNetwork()`. At some point we need to have a fallback. This place is where it is less "intrusive", i.e. cause less changes to other files.

See the other approach which touches lots of files: https://github.com/istio/istio/pull/49986